src/main/java/com/google/sps/servlets/UpdateMatrixServlet.java

```diff
@@ -58,10 +56,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
 			response.setContentType("application/json;");
 			response.getWriter().println(gson.toJson(updatedStringMatrix));
-		} 
-		catch (JSONException e) {
+		} catch (JSONException e) {
 		// crash and burn
+			Gson gson = new Gson();
+
+			response.setStatus(400);
+			response.getWriter().println(gson.toJson("Error parsing JSON request string"));
+
 			throw new IOException("Error parsing JSON request string");
+		} catch (NumberFormatException e) {
+			Gson gson = new Gson();
+
+			response.setStatus(400);
+			response.getWriter().println(gson.toJson("One or more probability is empty!"));
+
+			throw new IOException("One or more probability is empty!");
 		}
```

Comment:
From personal experience, it's  helpful to include the string that couldn't be parsed as part of the error message.